rts/Rendering/Fonts/CFontTexture.cpp

```diff
@@ -898,36 +1156,54 @@ void CFontTexture::LoadGlyph(std::shared_ptr<FontFace>& f, char32_t ch, unsigned
 	if (glyph.advance == 0 && glyph.size.w > 0)
 		glyph.advance = glyph.size.w;
 
-	const int width  = slot->bitmap.width;
-	const int height = slot->bitmap.rows;
+	int width  = slot->bitmap.width;
+	int height = slot->bitmap.rows;
 	const int olSize = 2 * outlineSize;
+	const int channels = slot->bitmap.pixel_mode == FT_PIXEL_MODE_BGRA ? 4 : 1;
 
-	if (width <= 0 || height <= 0)
-		return;
-
-	if (slot->bitmap.pixel_mode != FT_PIXEL_MODE_GRAY) {
+	if (slot->bitmap.pixel_mode != FT_PIXEL_MODE_GRAY && slot->bitmap.pixel_mode != FT_PIXEL_MODE_BGRA) {
 		LOG_L(L_ERROR, "invalid pixeldata mode");
 		return;
 	}
 
-	if (slot->bitmap.pitch != width) {
-		LOG_L(L_ERROR, "invalid pitch");
+	if (slot->bitmap.pitch != width*channels) {
+		LOG_L(L_ERROR, "invalid pitch %d %d", slot->bitmap.pitch, width);
```

Comment:
perhaps log channels as well since otherwise a colored font that is 4x too wide would appear correct in the log?
```suggestion
		LOG_L(L_ERROR, "invalid pitch %d (width %d channels %d)", slot->bitmap.pitch, width, channels);
```

Reply:
can do, but we're initializing `width` here from `slot->bitmap.width` a bit before, so I think channels won't really matter since it *has* to be slot->bitmap.pitch/channels, that's what we're testing.